packages/auth/src/core/strategies/passkey.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Auth, User, UserCredential } from '../../model/public_types';
+
+import {
+  startPasskeyEnrollment,
+  StartPasskeyEnrollmentRequest,
+  StartPasskeyEnrollmentResponse,
+  finalizePasskeyEnrollment,
+  FinalizePasskeyEnrollmentRequest,
+  FinalizePasskeyEnrollmentResponse,
+  startPasskeySignIn,
+  StartPasskeySignInRequest,
+  StartPasskeySignInResponse,
+  finalizePasskeySignIn,
+  FinalizePasskeySignInRequest,
+  FinalizePasskeySignInResponse,
+  publicKeyCredentialToJSON
+} from '../../api/account_management/passkey';
+import { UserInternal } from '../../model/user';
+import { _castAuth } from '../auth/auth_impl';
+import { getModularInstance } from '@firebase/util';
+import { OperationType } from '../../model/enums';
+import { UserCredentialImpl } from '../user/user_credential_impl';
+import { signInAnonymously } from './anonymous';
+
+const DEFAULT_PASSKEY_ACCOUNT_NAME = 'Unnamed account (Web)';
+const PASSKEY_LOOK_UP_ERROR_MESSAGE =
+  'The operation either timed out or was not allowed.';
+
+/**
+ * Signs in a user with a passkey.
```

Comment:
nit: let's add something like "Use enrollPasskey to enroll a passkey credential for the currently signed in user."